sync/optimistic.md

```diff
@@ -0,0 +1,290 @@
+justified checkpoint exists, any fork choice poisoning will be short-lived and
+resolved before that node is required to produce a block.
+
+However, the assumption that the honest, canonical chain will always justify
+within `SAFE_SLOTS_TO_IMPORT_OPTIMISTICALLY` slots is dubious. Therefore,
+clients MUST provide the following command line flag to assist with manual
+disaster recovery:
+
+- `--safe-slots-to-import-optimistically`: modifies the
+	`SAFE_SLOTS_TO_IMPORT_OPTIMISTICALLY`.
+
+## Checkpoint Sync (Weak Subjectivity Sync)
+
+A consensus engine MAY assume that the `ExecutionPayload` of a block used as an
+anchor for checkpoint sync is `VALID` without necessarily providing that
+payload to an execution engine.
+
+## Validator assignments
+
+An optimistic node is *not* a full node. It is unable to produce blocks, since
+an execution engine cannot produce a payload upon an unknown parent. It cannot
+faithfully attest to the head block of the chain, since it has not fully
+verified that block.
+
+### Block Production
+
+A optimistic validator MUST NOT produce a block (i.e., sign across the
+`DOMAIN_BEACON_PROPOSER` domain).
+
+### Attesting
+
+An optimistic validator MUST NOT participate in attestation (i.e., sign across the
+`DOMAIN_BEACON_ATTESTER`, `DOMAIN_SELECTION_PROOF` or
+`DOMAIN_AGGREGATE_AND_PROOF` domains).
+
+### Participating in Sync Committees
+
+An optimistic validator MUST NOT participate in sync committees (i.e., sign across the
+`DOMAIN_SYNC_COMMITTEE`, `DOMAIN_SYNC_COMMITTEE_SELECTION_PROOF` or
+`DOMAIN_CONTRIBUTION_AND_PROOF` domains).
+
+## P2P Networking
+
+### The Gossip Domain (gossipsub)
+
+#### `beacon_block`
+
+An optimistic validator MAY subscribe to the `beacon_block` topic. Propagation
+validation conditions are modified as such:
+
+Do not apply the existing condition:
+
+- [REJECT] The block's parent (defined by block.parent_root) passes validation.
+
+Instead, apply the new condition:
+
+- [REJECT] The block's parent (defined by block.parent_root) passes all
+    validation, excluding verification of the block.body.execution_payload.
+- [IGNORE] The block's parent (defined by block.parent_root) passes all
+    validation, including verification of the block.body.execution_payload.
+
+The effect of these modifications is that invalid payloads may be propagated
+across the network, but only when contained inside a block that is valid in *all
+other aspects*.
+
+#### Other Topics
+
+An optimistic node MUST NOT subscribe to the following topics:
+
+-  `beacon_aggregate_and_proof`
+-  `voluntary_exit`
+-  `proposer_slashing`
+-  `attester_slashing`
+-  `beacon_attestation_{subnet_id}`
+-  `sync_committee_contribution_and_proof`
+-  `sync_committee_{subnet_id}`
+
+Once the node ceases to be optimistic, it MAY re-subscribe to the
+aforementioned topics.
+
+### The Req/Resp Domain
+
+Non-faulty, optimistic nodes may send blocks which result in an INVALID
```

Comment:
Do we want this to be reflected in the `p2p-interface` as well?

Reply:
Also done in 1228e018.